app/main/views/frameworks.py

```diff
@@ -481,7 +481,7 @@ def upload_framework_agreement(framework_slug):
     try:
         email_body = render_template(
             'emails/framework_agreement_uploaded.html',
-            framework=framework,
+            framework_name=framework.get('name', 'Unknown framework'),
```

Comment:
I think this should probably be `framework['name']` if it doesn't have a name this should fail completely.